### PR TITLE
CASMNET-1637/CASMNET-1368 - Improve documentation for updating the `external-dns` IP address.

### PR DESCRIPTION
**Pull request by spillerc-hpe**
# Description

* Added steps to update `customizations.yaml` and SLS to the procedure to change the `external-dns` IP address.
* Added a link to the `externap-dns` IP address update procedure to the SLS Upgrade documentation.

Addresses the following JIRA tickets.

[CASMNET-1637](https://jira-pro.its.hpecorp.net:8443/browse/CASMNET-1637) - Document process to change system_to_site_lookups if CAN isn't converted to CMN
[CASMNET-1638](https://jira-pro.its.hpecorp.net:8443/browse/CASMNET-1638) - Add steps to update customizations and SLS to procedure to change external DNS IP

# Checklist Before Merging

- [X] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [X] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [X] My commits or Pull-Request Title contain my JIRA information, or I don't have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/MTL-1695/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

**Review by dle-hpe — APPROVED** [review]

File: operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md [hunk 1]
```diff
@@ -27,16 +29,79 @@ The system is installed.
...
+
+   ```json
+   {
+     "Comment": "site to system lookups",
+     "IPAddress": "x.x.x.x",
+     "Name": "external-dns"
+   }
+   ```
+
+1. Upload the updated `NMN.json` file to SLS.
```
Comment (nrockershousen) [inline operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md:75]:
```suggestion
1. Upload the updated `CMN.json` file to SLS.
```
I assume this should be CMN?

File: operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md [hunk 2]
```diff
@@ -27,16 +29,79 @@ The system is installed.
...
-    2. Edit the `cray-dns-powerdns-cmn-udp` service.
+    1. Edit the `cray-dns-powerdns-cmn-udp` service.
 
         ```bash
         ncn-m001# kubectl -n services edit svc cray-dns-powerdns-cmn-udp
         ```
+
+### Update SLS
+
+The `external-dns` IP reservation in the SLS `CMN` network `cmn_metallb_static_pool` subnet should be updated to the desired CMN IP address.
```
Comment (mharding-hpe) [inline operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md:48]:
```suggestion
The `external-dns` IP address reservation in the SLS CMN `cmn_metallb_static_pool` subnet should be updated to the desired CMN IP address.
```

File: operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md [hunk 3]
```diff
@@ -27,16 +29,79 @@ The system is installed.
...
+
+1. Retrieve the SLS data for `CMN` network.
```
Comment (mharding-hpe) [inline operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md:50]:
```suggestion
1. Retrieve the SLS data for CMN.
```

File: operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md [hunk 4]
```diff
@@ -27,16 +29,79 @@ The system is installed.
...
+
+   ```bash
+   ncn-m001# export TOKEN=$(curl -s -k -S -d grant_type=client_credentials \
+   -d client_id=admin-client -d client_secret=`kubectl get secrets admin-client-auth \
+   -o jsonpath='{.data.client-secret}' | base64 -d` \
```
Comment (mharding-hpe) [inline operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md:55]:
```suggestion
                 -o jsonpath='{.data.client-secret}' | base64 -d` \
```

File: operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md [hunk 5]
```diff
@@ -27,16 +29,79 @@ The system is installed.
...
+   https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token \
```
Comment (mharding-hpe) [inline operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md:56]:
```suggestion
                 https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token \
```

File: operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md [hunk 6]
```diff
@@ -27,16 +29,79 @@ The system is installed.
...
+   | jq -r '.access_token')
```
Comment (mharding-hpe) [inline operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md:57]:
```suggestion
                 | jq -r '.access_token')
```

File: operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md [hunk 7]
```diff
@@ -27,16 +29,79 @@ The system is installed.
...
+
+   ncn-m001# curl -s -k -H "Authorization: Bearer ${TOKEN}" \
+   https://api-gw-service-nmn.local/apis/sls/v1/networks/CMN|jq > CMN.json
```
Comment (mharding-hpe) [inline operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md:60]:
```suggestion
                 https://api-gw-service-nmn.local/apis/sls/v1/networks/CMN|jq > CMN.json
```

File: operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md [hunk 8]
```diff
@@ -27,16 +29,79 @@ The system is installed.
...
+     "IPAddress": "x.x.x.x",
+     "Name": "external-dns"
+   }
+   ```
+
+1. Upload the updated `NMN.json` file to SLS.
+
+   ```bash
+   ncn-m001# curl -s -k -H "Authorization: Bearer ${TOKEN}" --header \
+   "Content-Type: application/json" --request PUT --data @CMN.json \
```
Comment (mharding-hpe) [inline operations/network/external_dns/Update_the_cmn-external-dns_Value_Post-Installation.md:79]:
```suggestion
                 "Content-Type: application/json" --request PUT --data @CMN.json \
```

File: upgrade/1.2/scripts/sls/README.SLS_Upgrade.md [hunk 9]
```diff
@@ -19,12 +19,17 @@ During the update of SLS, at a minimum, answers to the following questions must
      **WARNING:** It is up to the user to compare pre-upgraded and post-upgraded SLS files for sanity. Specifically, in the
      case of preserving `external-dns` values, to prevent site-networking changes that might result in NCN IP addresses
      overlapping during the upgrade process. This requires network subnetting expertise and EXPERT mode below.
+
+     If the `external-dns` IP address is changed the `customizations.yaml` `site_to_system_lookups` value needs to be updated to
```
Comment (mharding-hpe) [inline upgrade/1.2/scripts/sls/README.SLS_Upgrade.md:23]:
```suggestion
     If the `external-dns` IP address is changed, then the `customizations.yaml` `site_to_system_lookups` value must be updated to
```

**Review by mharding-hpe — APPROVED** [review]

**Review by nrockershousen — APPROVED** [review]

**Pull request merged**